### DISCUSSION TITLE
defaultValue for unused field

### DISCUSSION THREAD

**Discussion by ngregrichardson** [edmundhung/conform#821] [Q&A] [answered]
Hi,

First of all, I've absolutely loved using Conform. It has made life👨‍🍳 💋 .

However, I have one page of my NextJS app that takes in a query parameter that must be submitted. This is how my component is defined:

```ts
const resetPasswordSchema = z.object({
    password: z.string(),
    confirmPassword: z.string(),
    token: z.string()
}).refine((data) => data.password === data.confirmPassword, {
    message: "Passwords must match",
    path: ["confirm"]
});
```

```ts
const [lastResult, action, isPending] = useActionState(resetPassword, undefined);
    const [form, fields] = useForm({
        lastResult,
        defaultValue: {
            token
        },
        onValidate: ({ formData }) => parseWithZod(formData, { schema: resetPasswordSchema }),
        shouldValidate: 'onBlur',
        shouldRevalidate: 'onInput'
    });
```

`token` is passed in as a prop, and there is no input element for `token` on the page. However, when I fill out the rest of my form, it fails with a validation error silently (since there is no input to show the error. Not too sure why, I feel like I'm just missing something obvious. Any ideas? Is this expected? Thanks!

![image](https://github.com/user-attachments/assets/1b18f9e3-b209-4f63-ac04-1f344b67ce29)

**Comment by edmundhung** [comment, marked as answer]
> token is passed in as a prop, and there is no input element for token on the page.

That’s likely the issue. Server actions rely on `FormData`, which reads all inputs associated with your form directly from the DOM. Since Conform doesn’t automatically populate any input elements, you’ll need to render a hidden input for the token manually.

You can verify this by logging the formData and submission result in the `onValidate` handler like this:
```tsx
const [form, fields] = useForm({
    lastResult,
    defaultValue: {
        token
    },
    onValidate: ({ formData }) => {
        const submission = parseWithZod(formData, { schema });

        console.log({
            formData: Object.fromEntries(formData),
            result: submission.reply(),
        });

        return submission
    },
    shouldValidate: 'onBlur',
    shouldRevalidate: 'onInput'
});
```

Hope this helps!

**Reply by ngregrichardson** [reply]
🤦 that makes total sense, I'm not really sure why I didn't think about it. Thanks!!
